browser/ui/views/split_view/split_view.cc

```diff
@@ -0,0 +1,408 @@
+
+// A `ContentsWebView` that activates its contents when it gets focus.
+class ActivatableContentsWebView : public ContentsWebView {
+  METADATA_HEADER(ActivatableContentsWebView, ContentsWebView)
+ public:
+  using ContentsWebView::ContentsWebView;
+  ~ActivatableContentsWebView() override = default;
+
+  // ContentsWebView:
+  void OnFocus() override {
+    ContentsWebView::OnFocus();
+    if (web_contents() && web_contents()->GetDelegate()) {
+      web_contents()->GetDelegate()->ActivateContents(web_contents());
+    }
+  }
+};
+BEGIN_METADATA(ActivatableContentsWebView)
+END_METADATA
+
+}  // namespace
+
+SplitView::SplitView(Browser* browser,
+                     views::View* contents_container,
+                     ContentsWebView* contents_web_view)
+    : browser_(browser),
+      contents_container_(contents_container),
+      contents_web_view_(contents_web_view) {
+  CHECK(base::FeatureList::IsEnabled(tabs::features::kBraveSplitView));
+
+  // Re-parent the |contents_container| to this view.
+  AddChildView(
+      contents_container->parent()->RemoveChildViewT(contents_container));
+
+  // Make secondary contents view and related views to support split view mode
+  secondary_contents_container_ = AddChildView(std::make_unique<views::View>());
+  secondary_contents_container_->SetVisible(false);
+
+  secondary_devtools_web_view_ = secondary_contents_container_->AddChildView(
+      std::make_unique<views::WebView>(browser_->profile()));
+  secondary_contents_web_view_ = secondary_contents_container_->AddChildView(
+      std::make_unique<ActivatableContentsWebView>(browser_->profile()));
+
+  split_view_separator_ =
+      AddChildView(std::make_unique<SplitViewSeparator>(browser_.get()));
+
+  secondary_contents_container_->SetLayoutManager(
+      std::make_unique<ContentsLayoutManager>(secondary_devtools_web_view_,
+                                              secondary_contents_web_view_));
+
+  SetLayoutManager(std::make_unique<SplitViewLayoutManager>(
+      contents_container_, secondary_contents_container_,
+      split_view_separator_));
+
+  auto* split_view_browser_data = SplitViewBrowserData::FromBrowser(browser);
+  GetSplitViewLayoutManager()->set_split_view_browser_data(
+      split_view_browser_data);
+  split_view_observation_.Observe(split_view_browser_data);
+}
+
+SplitView::~SplitView() = default;
+
+SplitView::AfterSwapWebContents SplitView::WillSwapActiveWebContents(
+    BrowserViewKey,
+    content::WebContents* new_contents,
+    int index) {
+  bool need_to_update_secondary_web_view = false;
+  // In order to minimize flickering during tab activation, we should update
+  // split view only when it's needed.
+  auto* browser_data = SplitViewBrowserData::FromBrowser(browser_.get());
+  auto* tab_strip_model = browser_->tab_strip_model();
+  if (auto tile =
+          browser_data->GetTile(tab_strip_model->GetTabHandleAt(index))) {
+    auto* main_web_contents = tab_strip_model->GetWebContentsAt(
```

Comment:
I don't think we should be swapping the pointer in the tab strip, that could cause all kinds of potential side effects. Why didn't you just pass `devtools_web_view_` and `contents_web_view_` into the constructor so it can work like it did before?

Reply:
We don't swap it from tab strip. Maybe `WillSetActiveWebContentsToContentsWebView` would be better name.